File: src/go/cliParams.go
package main

import "os"
import "fmt"

func main() {
	if len(os.Args) == 1 {
		fmt.Println("No parameters")
	} else if len(os.Args) == 2 {
		fmt.Println("Parameter is " + os.Args[1])
	} else {
		fmt.Println("Too many parameters")
	}
}

File: src/cpp/cliParams.cpp
#include <iostream>

int main(int argc, char *argv[])
{
    if (argc == 1)
        std::cout << "No parameters";
    else if (argc == 2)
        std::cout << "Parameter is " << argv[1];
    else
        std::cout << "Too many parameters";

    return 0;
}